### PR TITLE
feat: Drop strict mode in favor of leniency flags.

### PR DESCRIPTION
**Pull request by ShogunPanda**
This PR drops completely the strict mode in favor of three new leniency flags.

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

does this impact the benchmarks at all?

**Comment by ShogunPanda** [comment]
Nope, nothing at all.

Before:

```
> llhttp@8.1.1 bench /Volumes/DATI/Users/Shogun/Programmazione/Work/NearForm/internal/llhttp
> ts-node bench/

url loose (C)
8192.00 mb | 2559.11 mb/s | 51604286.45 ops/sec | 3.20 s
url strict (C)
8192.00 mb | 2328.31 mb/s | 46950162.77 ops/sec | 3.52 s
http loose: "seanmonstar/httparse" (C)
8192.00 mb | 1690.68 mb/s | 2521771.31 ops/sec | 4.85 s
http strict: "seanmonstar/httparse" (C)
8192.00 mb | 1693.59 mb/s | 2526117.23 ops/sec | 4.84 s
http loose: "nodejs/http-parser" (C)
8192.00 mb | 1375.00 mb/s | 2772670.79 ops/sec | 5.96 s
http strict: "nodejs/http-parser" (C)
8192.00 mb | 1400.48 mb/s | 2824064.73 ops/sec | 5.85 s
```

After (consider that the new mode is equivalent to the old strict mode):

```
> llhttp@8.1.1 bench /Volumes/DATI/Users/Shogun/Programmazione/Work/NearForm/internal/llhttp
> ts-node bench/

url (C)
8192.00 mb | 2334.68 mb/s | 47078602.73 ops/sec | 3.51 s
http: "seanmonstar/httparse" (C)
8192.00 mb | 1694.82 mb/s | 2527946.41 ops/sec | 4.83 s
http: "nodejs/http-parser" (C)
8192.00 mb | 1379.50 mb/s | 2781756.35 ops/sec | 5.94 s
```

Are we good to merge and release?

File: README.md [hunk 1]
```diff
@@ -333,6 +333,47 @@ With this flag the extra value will be parsed normally.
...
 
+### `void llhttp_set_lenient_version(llhttp_t* parser, int enabled)`
+
+Enables/disables lenient handling of HTTP version.
+
+Normally `llhttp` would error when the HTTP version in the request or status line
+is not `0.9`, `1.0`, `1.1` or `2.0`.
+With this flag the extra value will be parsed normally.
+
+**(USE AT YOUR OWN RISK)**
```
Comment (RafaelGSS) [inline README.md:344]: I'd rephrase it to something more specific... like, if you are relying on this parsing, don't use this function from a security point of view. Or mention some CWE it implies.

**Review by RafaelGSS — COMMENTED** [review]
It's missing the breaking change label, right? Also, would be great to have a PR description so people can look at it when evaluating llhttp 9

File: README.md (hunk 1 above)
Comment (ShogunPanda) [reply README.md]: What about now?

**Review by RafaelGSS — APPROVED** [review]

**Comment by davidbolvansky** [comment]
For people coming from old http_parser it would be good to write steps how to migrate to llhttp if they used STRICT=OFF in cmake.
